mem/interface.md

```diff
@@ -82,9 +82,9 @@ pub struct Pointer<Provenance> {
 /// executing the same operation on the same memory can have different results.
 /// We also let read operations potentially mutate memory (they actually can
 /// change the current state in concurrent memory models and in Stacked Borrows).
-pub trait Memory: Sized {
+pub trait Memory: specr::Obj {
     /// The type of pointer provenance.
-    type Provenance: Eq + Hash + Clone + Debug;
+    type Provenance: specr::Obj + Eq + Hash;
```

Comment:
`Eq` and `Hash` should not be needed any more, after our discussion? :)

(Again would be nice to just add the bounds automatically. They will look pretty different in Coq.)